JustSaying/Messaging/PublishEnvelope.cs

```diff
@@ -15,6 +15,25 @@ public PublishEnvelope(Message message)
 
         public TimeSpan? Delay { get; set; }
 
-        public IDictionary<string, MessageAttributeValue> MessageAttributes { get; } = new Dictionary<string, MessageAttributeValue>();
+        public IDictionary<string, MessageAttributeValue> MessageAttributes { get; set; }
+
+        public void AddMessageAttribute(string key, MessageAttributeValue value)
+        {
+            if (MessageAttributes == null)
+            {
+                MessageAttributes = new Dictionary<string, MessageAttributeValue>();
+            }
+
+            MessageAttributes.Add(key, value);
```

Comment:
Suggest using `[]` here instead of `Add()`, as if the value got added twice you'd get an `ArgumentException` because the key was already there.